qa-pipelines/qa-pipeline.yml.erb

```diff
@@ -0,0 +1,268 @@
+    else "https://((s3-config-bucket-sles)).s3.amazonaws.com/((s3-config-prefix-sles))#{ERB::Util.url_encode(version)}.zip"
+  end
+end
+
+%>
+---
+# Concourse pipeline to deploy and upgrade CAP
+
+resource_types:
+- name: github-status
+  type: docker-image
+  source:
+    repository: resource/github-status
+    tag: release
+
+resources:
+# USB sidecar archive for postgres
+- name: s3.pg-sidecar
+  type: s3
+  source:
+    endpoint: ((s3-config-endpoint))
+    access_key_id: ((s3-config-access-key))
+    secret_access_key: ((s3-config-secret-key))
+    bucket: ((s3-config-bucket-sles))
+    regexp: ((s3-config-prefix-sidecars))cf-usb-sidecar-postgres-(.*)\.tgz$
+
+# USB sidecar archive for MySQL
+- name: s3.mysql-sidecar
+  type: s3
+  source:
+    endpoint: ((s3-config-endpoint))
+    access_key_id: ((s3-config-access-key))
+    secret_access_key: ((s3-config-secret-key))
+    bucket: ((s3-config-bucket-sles))
+    regexp: ((s3-config-prefix-sidecars))cf-usb-sidecar-mysql-(.*)\.tgz$
+
+# CI helper scripts repository
+- name: ci
+  type: git
+  source:
+    uri: ((src-ci-repo))
+    branch: ((src-ci-branch))
+    paths:
+    - qa-pipelines/*
+    - sample-apps/*
+
+# SCF bundle to install from (upgrade target, if upgrading)
+- name: s3.scf-config-sles
+  type: s3
+  source:
+    endpoint: ((s3-config-endpoint))
+    access_key_id: ((s3-config-access-key))
+    secret_access_key: ((s3-config-secret-key))
+    bucket: ((s3-config-bucket-sles))
+    regexp: ((s3-config-prefix-sles))scf-sle-(.*)\.zip$
+
+# Pool resource with kube cluster information
+- name: pool.kube-hosts
+  type: pool
+  source:
+    uri: ((kube-pool-repo))
+    private_key: ((kube-pool-key))
+    branch: ((kube-pool-branch))
+    pool: ((kube-pool-pool))
+
+# GitHub commit status
+<% if status_reporting %>
+- name: status.src
+  type: github-status
+  source:
+    repo: ((src-repo))
+    access_token: ((github-access-token))
+<% end %>
+
+jobs:
+<% %w(SA HA).each do |avail| %>
+<% next if avail == 'HA' and not ha %>
+<% next if avail == 'SA' and not sa %>
+- name: ((pipeline-name))-<%= avail %>
+  plan:
+  - do:
+    - aggregate:
+      - get: ci
+      - get: s3.pg-sidecar
+        trigger: true
+      - get: s3.mysql-sidecar
+        trigger: true
+      - get: s3.scf-config-sles
+        trigger: true
+      - put: pool.kube-hosts
+        params: {acquire: true}
+        timeout: 2m
+    - task: cf-get-commit-id
+      file: ci/qa-pipelines/tasks/cf-get-commit-id.yml
+      input_mapping:
+        s3.archive: s3.scf-config-sles
+    <% if status_reporting %>
+    - put: status.src
+      params: &status-params-<%= avail %>
+        context: ((pipeline-name))-<%= avail %>
+        description: "QA Pipeline: ((pipeline-name)) (SLES <%= avail %>)"
+        path: commit-id/sha
+        state: pending
+    <% end %> # status_reporting
+    on_failure:
+      put: pool.kube-hosts
+      params: {release: pool.kube-hosts}
+  - do:
+    <% if upgrade_from_version %>
```

Comment:
Can you add an inline comment clarifying that we populate this part of the template for upgrade pipelines? But actually, I'm not sure where this is supposed to be set. In our existing config files, we use the parameter `cap-sle-url`

Reply:
Talked about this, it might be better to consolidate the config files into one (which will include the pre-upgrade bundle URL) and have the set-pipeline script add the necessary changes to the pool-specific parameters. set-pipeline can also take a flag to override the *deploy* version for deployments from custom versions (nightlies, PRs, etc.). The pre-upgrade bundle URL was previously being used for this in a hacky way, but the refactor can improve all this